core/src/main/java/org/infinispan/remoting/inboundhandler/BasePerCacheInboundInvocationHandler.java

```diff
@@ -195,6 +196,7 @@ protected boolean beforeInvoke() throws Exception {
                handler.getLog().tracef("Ignoring command sent before the local node was a member " +
                                              "(command topology id is %d)", commandTopologyId);
             }
+            response = CacheNotFoundResponse.INSTANCE;
```

Comment:
I wonder if making this more explicit would be cleaner.  What if the beforeInvoke method declaration was changed to  return a Response.  If that value was null it would be considered to continue running and if not null it would return that response?  WDYT?